ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py

```diff
@@ -2557,3 +2562,47 @@ def test_torch_instance_pow_(dtype_and_x, as_variable, native_array):
         class_name="tensor",
         method_name="pow_",
     )
+
+
+@st.composite
+def _get_dtype_and_multiplicative_matrices(draw):
+    return draw(
+        st.sampled_from(
+            [
+                _get_dtype_input_and_matrices(),
+                _get_dtype_and_3dbatch_matrices(),
+                _get_dtype_input_and_mat_vec(),
+            ]
+        )
+    )
+
+
+# matmul
+@handle_cmd_line_args
+@given(
+    dtype_tensor1_tensor2=_get_dtype_and_multiplicative_matrices().example(),
+)
+def test_torch_instance_matmul(
+    dtype_tensor1_tensor2,
+    as_variable,
```

Comment:
where are `as_variable` and `native_array` defined?

Reply:
The `@handle_cmd_line_args` decorator takes care of these strategies at runtime so there's no need to explicitly define them in the test. I have never had to define them previously whenever I have contributed to the codebase. Additionally, you can read point no. 3 [here](https://github.com/unifyai/ivy/pull/5150#issuecomment-1264541807). One of the maintainers of this repo actually advised me against defining strategies for `as_variable` and `native_array` and let the `@handle_cmd_line_args` take care of that for us. I don't think anyone else has defined them explicitly either.

Reply:
Hi @hmahmood24, indeed you're correct, my apologies, I haven't been working on the frontends so I'm a little behind with some of the strategies updates.

As for the the assertion error, perhaps try adjusting the `rtol` and see if that makes a difference, although I'm concerned about the tolerance being high already. This issue is surprising, especially given the ground truth reference is torch also.